py-polars/polars/utils/_construction.py

```diff
@@ -664,6 +665,11 @@ def _post_apply_columns(
     return pydf
 
 
+@functools.lru_cache(maxsize=128)
```

Comment:
Do we really need to add another method here?

Why can't you put the cache on the `is_polar_dtype` method? Is there any reason for having 2 different methods doing exactly the same?

That would make this PR a 2-line change (import + decorator on the original method), solving the issue as well?

Reply:
> Do we really need to add another method here?

That is a valid question. My motivation is that this cache speeds up calls to `pl.from_records()` by ~10% in my use case.
 
> Why can't you put the cache on the `is_polar_dtype` method? Is there any reason for having 2 different methods doing exactly the same?

`is_polars_dtype()` may be called with non-hashable arguments in other code paths, breaking the decorated function. I experimented with a holistic solution in #11054, but the try-except overhead appears to outweigh any performance gains.

Reply:
> `is_polars_dtype()` may be called with non-hashable arguments in other code paths, breaking the decorated function. I experimented with a holistic solution in #11054, but the try-except overhead appears to outweigh any performance gains.

Thanks for the response. I didn't consider these indeed...

Reply:
@jakob-keller can you add a comment on the rationale of having 2 methods.

Reply:
> @jakob-keller can you add a comment on the rationale of having 2 methods.

Will do

Reply:
Have a look at #11068 for a more comprehensive approach.